particles/Tutorial/Kotlin/README.md

```diff
@@ -212,3 +212,367 @@ arcs_kt_binary(
+use these concepts, so you will have plenty more examples to contemplate.
+
+# The Template Interpolation Revisitation
+So it’s time for a small confession. We didn’t give the full picture of how handles work in the previous section. 
+To understand the full picture, we need a new definition, and to update our definition of handles:
+
+>- *Stores* - A store represents a data location
+>- *Handles* - Handles are manifestations of stores inside an arc. They allow particles to read, write and listen for data updates.
+
+In essence, a store is where the data physically exists. The handle lets different particles access this data. With this understanding, we can update the picture we introduced in the last tutorial to include the store.
+
+![Store Image](../screenshots/store-diagram.png)
+
+In the last tutorial, we set the Person handle inside of the particles. Person was also a singleton, it represented a single instance of the Person schema. In this tutorial, we will instead define a collection with multiple instances of Person inside the Arcs Manifest file. While a collection can be thought of as a list, there are no guarantees about the order of elements within the collection.
+Using some more sophisticated template interpolation, we can easily greet everyone in our collection. But that’s enough explanation, let’s get to some code!
+
+We begin with the Arcs manifest file which includes the collection of `PersonDetails`.
+```
+schema PersonDetails
+  name: Text
+  age: Number
+
+// This is essentially a JSON file defined inside the manifest.
+resource PeopleData
+  start
+  [
+    {"name": "Jill", "age": 70},
+    {"name": "Jack", "age": 25},
+    {"name": "Jen", "age": 50}
+  ]
+
+// This data store contains a collection of entities rather than a single
+// entity, and is backed by the PeopleData resource defined above.
+store PeopleToGreetStore of [PersonDetails] in PeopleData
+
+particle CollectionsParticle in 'collections.wasm'
+  // The input is a collection of PersonDetails entities.
+  inputData: reads [PersonDetails]
+  root: consumes
+
+recipe CollectionRecipe
+  data: map PeopleToGreetStore
+
+  CollectionsParticle
+    inputData: reads data
+
+  description `Kotlin Tutorial 5: Collections`
+```
+
+And the Kotlin:
+```kotlin
+package arcs.tutorials
+
+import arcs.*
+import kotlin.native.internal.ExportForCppRuntime
+
+/**
+ * Sample Kotlin-WASM Particle to use a JSON store.
+ */
+class CollectionsParticle : Particle() {
+    // Note this is a Collection handle instead of a Singleton.
+   private val people = Collection { CollectionsParticle_InputData() }
+
+   init {
+       registerHandle("inputData", people)
+   }
+
+    override fun populateModel(slotName: String, model: Map<String, Any?>): Map<String, Any?> {
+        // We begin by generating the list of models that should fill the template. Our template has name ang age so we
```

Comment:
Might be good to limit the column width of these examples to something like 100 or 120 chars.